samza-core/src/main/java/org/apache/samza/startpoint/StartpointManager.java

```diff
@@ -0,0 +1,273 @@
+
+    byte[] startpointBytes = metadataStore.get(toStoreKey(ssp, taskName));
+
+    return Objects.isNull(startpointBytes) ? null : startpointSerde.fromBytes(startpointBytes);
+  }
+
+  /**
+   * Deletes the {@link Startpoint} for a {@link SystemStreamPartition}
+   * @param ssp The {@link SystemStreamPartition} to delete the {@link Startpoint} for.
+   */
+  public void deleteStartpoint(SystemStreamPartition ssp) {
+    Preconditions.checkNotNull(metadataStore, "Underlying metadata store not available");
+    Preconditions.checkNotNull(ssp, "SystemStreamPartition cannot be null");
+
+    metadataStore.delete(toStoreKey(ssp));
+  }
+
+  /**
+   * Deletes the {@link Startpoint} for a {@link SystemStreamPartition} and {@link TaskName}.
+   * @param ssp ssp The {@link SystemStreamPartition} to delete the {@link Startpoint} for.
+   * @param taskName ssp The {@link TaskName} to delete the {@link Startpoint} for.
+   */
+  public void deleteStartpointForTask(SystemStreamPartition ssp, TaskName taskName) {
+    Preconditions.checkNotNull(metadataStore, "Underlying metadata store not available");
+    Preconditions.checkNotNull(ssp, "SystemStreamPartition cannot be null");
+    Preconditions.checkNotNull(taskName, "TaskName cannot be null");
+    Preconditions.checkArgument(StringUtils.isNotBlank(taskName.getTaskName()), "TaskName cannot be blank");
+
+    metadataStore.delete(toStoreKey(ssp, taskName));
+  }
+
+  /**
+   * For {@link Startpoint}s keyed only by {@link SystemStreamPartition}, this method remaps the Startpoints for the specified
+   * SystemStreamPartition to SystemStreamPartition+{@link TaskName} for all tasks provided by the {@link SystemStreamPartitionGrouper}
+   * This method is not atomic or thread-safe. The intent is for the Samza Processor's coordinator to use this
+   * method to assign the Startpoints to the appropriate tasks.
+   * @param ssp The {@link SystemStreamPartition} that a {@link Startpoint} is keyed to.
+   * @param grouper The {@link SystemStreamPartitionGrouper} is used to determine what the task names are.
+   * @return The list of {@link TaskName}s
+   */
+  public Set<TaskName> groupStartpointsPerTask(SystemStreamPartition ssp, SystemStreamPartitionGrouper grouper) {
+    Preconditions.checkNotNull(metadataStore, "Underlying metadata store not available");
+    Preconditions.checkNotNull(ssp, "SystemStreamPartition cannot be null");
+    Preconditions.checkNotNull(grouper, "SystemStreamPartitionGrouper cannot be null");
+
+    Startpoint startpoint = readStartpoint(ssp);
+    HashMap<TaskName, Startpoint> result = new HashMap<>();
+
+    LOG.info("Grouping Startpoint keyed on SSP: {} to tasks determined by the grouper: {}", ssp, grouper.getClass().getCanonicalName());
+
+    // Re-map startpoints from SSP-only keys to SSP+TaskName keys
+    Map<TaskName, Set<SystemStreamPartition>> group = grouper.group(Sets.newHashSet(ssp));
```

Comment:
1. Why do we need the grouper here to generate the task name, because kafka checkpoint  [key](https://github.com/apache/samza/blob/master/samza-kafka/src/main/java/org/apache/samza/checkpoint/kafka/KafkaCheckpointLogKey.java) needs it?
2. What if the grouper is changed by the user after publishing the startpoint and before the subsequent deployment.